horace_core/sqw/@Experiment/private/build_from_old_headers_.m

```diff
@@ -40,45 +65,91 @@
             class(instr),i)
     end
     
+    % Locally name the sample (again, not liking plural samples, same reason)
     sampl = hdr.samples;
-    %
-    if isstruct(sampl) && isempty(fieldnames(sampl))
-        %TODO: should it be sample factory?, returning samples as function
-        % of inputs. The same as instrument
-        sampl = IX_samp();
-        sampl.alatt = alatt;
-        sampl.angdeg = angdeg;
-    elseif isa(sampl,'IX_samp') && isempty(sampl)
+    
+    % If sampl is a struct
+    if isstruct(sampl)
+        % Sample in header is an empty struct, replace with null sample and
+        % populate with the lattice parameters
+        if isempty(sampl) || isempty(fieldnames(sampl))
+            sampl = IX_null_sample();
```

Comment:
null_sample means sqw object is broken. Have never happened and should not in a future.

Reply:
Nonetheless, defending against it and being able to catch it is useful.

Reply:
Samples frequently occur in the existing data from rundata, files etc with an empty values or with a fieldless struct. The input data did not have the sample info and presumed that the sample would be defined later. These lines formalize that by making an IX_null_sample to mark the existence of this variable (sample) as a sample with no data yet applied.